src/main/java/crawlercommons/sitemaps/AbstractSiteMap.java

```diff
@@ -53,6 +53,38 @@ protected DateFormat initialValue() {
         }
     };
 
+    private static final ThreadLocal<DateFormat> W3C_FULLDATE_FORMAT_WITH_OFFSET = new ThreadLocal<DateFormat>() {
+        protected DateFormat initialValue() {
+            SimpleDateFormat result = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ", Locale.ROOT);
+            result.setTimeZone(TimeZone.getTimeZone("UTC"));
+            return result;
+        }
+    };
+
+    protected static final ThreadLocal<DateFormat> RFC822_FULLDATE_FORMAT = new ThreadLocal<DateFormat>() {
+        protected DateFormat initialValue() {
+            return new SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss Z", Locale.ROOT);
+        }
+    };
+
+    protected static final ThreadLocal<DateFormat> RFC822_NO_WEEKDAY_DATE_FORMAT = new ThreadLocal<DateFormat>() {
+        protected DateFormat initialValue() {
+            return new SimpleDateFormat("dd MMM yyyy HH:mm:ss Z", Locale.ROOT);
+        }
+    };
+
+    protected static final ThreadLocal<DateFormat> RFC822_SHORT_YEAR_DATE_FORMAT = new ThreadLocal<DateFormat>() {
+        protected DateFormat initialValue() {
+            return new SimpleDateFormat("EEE, dd MMM yy HH:mm:ss Z", Locale.ROOT);
+        }
+    };
+
+    protected static final ThreadLocal<DateFormat> RFC822_SHORTDATE_FORMAT = new ThreadLocal<DateFormat>() {
```

Comment:
Actually only 4. The fifth one is for generating string representation for parsing by convertToDate. Not sure if it is necessary to cover 'with and without timezone'.
Also, probably it could be reasonable to use single ThreadLocal<DateFormat[]> instead of many new static fields.